src/components/Cardgrid.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import {
+  CircularProgress,
+  FontIcon,
+  IconButton,
+} from 'material-ui';
+import {
+  GridList,
+  GridTile
+} from 'material-ui/GridList';
+import { muiThemeable } from 'material-ui/styles';
+
+const styles = {
+  image: {
+    width: 'initial',
+    height: 'initial',
+    maxWidth: '100%',
+    maxHeight: '100%',
+  },
+  load: {
+    width: '100%',
+    height: 135,
+    paddingTop: 40,
+    boxSizing: 'border-box',
+    backgroundColor: '#606060',
+    textAlign: 'center',
+  },
+  loadmore: {
+    color: 'white',
+    fontSize: 30,
+    cursor: 'pointer',
+    textDecoration: 'underline',
+    backgroundColor: 'transparent',
+    border: 'none',
+    outline: 'none',
+  }
+};
+
+export default muiThemeable()(Cardgrid);
+
+function Cardgrid({
+  data,
+  loadMore,
+  loading,
+  muiTheme,
+}) {
+  return (
+    <div style={{fontFamily: muiTheme.fontFamily}}>
+      <div style={muiTheme.container}>
+        <GridList cellHeight={300}>
+          {data.map(item => (
```

Comment:
let's move this into a separate stateless functional component. I read somewhere that multiline TSX expressions were buggy. It is actually cleaner to create a function or stateless function component, and assign it, or the result, into a variable.
```
const actionIconButton = item => (
  <IconButton>
    <FontIcon
      color="white"
      hoverColor="#AD000D"
      className='fa fa-fire'
    />
  </IconButton>)
const mappedTiles = data.map(item => (
  <GridTile
    key={item.key}
    title={item.title}
    subtitle={<span><b>{item.price} - {item.approx}</b></span>}
    actionIcon={actionIconButton(item)}
  >
    <img style={styles.image} src={item.img} alt={item.title} />
  </GridTile>))
```
```
return (
  // ...
  <GridList cellHeight={300}>
    {mappedTiles}
  </GridList>
```